### PR TITLE
Fix two issues related to infinite Range objects

### LINKED ISSUE #17857: Pretty print Range(-oo, oo) raises
```python
>>> from sympy import *
>>> init_printing()
>>> Range(-oo, oo)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/enojb/current/sympy/sympy/sympy/interactive/printing.py", line 30, in _displayhook
    print(stringify_func(arg, **settings))
  File "/Users/enojb/current/sympy/sympy/sympy/printing/pretty/pretty.py", line 2606, in pretty
    return pp.doprint(expr)
  File "/Users/enojb/current/sympy/sympy/sympy/printing/pretty/pretty.py", line 64, in doprint
    return self._print(expr).render(**self._settings)
  File "/Users/enojb/current/sympy/sympy/sympy/printing/printer.py", line 287, in _print
    return getattr(self, printmethod)(expr, **kwargs)
  File "/Users/enojb/current/sympy/sympy/sympy/printing/pretty/pretty.py", line 1918, in _print_Range
    printset = dots, s[-1] - s.step, s[-1]
  File "/Users/enojb/current/sympy/sympy/sympy/sets/fancysets.py", line 842, in __getitem__
    raise ValueError(ooslice)
ValueError: cannot slice from the end with an infinite value
```

### LINKED ISSUE #17858: Range(-oo, oo) doesn't allow any membership checking
```
>>> from sympy import *
>>> 1 in Range(-oo, oo)
False
>>> 1 in Range(-oo, oo, 1)
False
```

### PR DESCRIPTION
**Pull request by ethankward**
#### References to other Issues or PRs
Fixes #17858. Fixes #17857.

#### Brief description of what is fixed or changed
Check if both sides are infinite and print `{..., -1, 0, 1, ...}`.

Fix membership checking if both sides are infinite.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by smichr** [comment]
```diff
diff --git a/sympy/sets/fancysets.py b/sympy/sets/fancysets.py
index 5a1cf8d..6c8f863 100644
--- a/sympy/sets/fancysets.py
+++ b/sympy/sets/fancysets.py
@@ -659,7 +659,13 @@ def _contains(self, other):
                 _ = self.size  # validate
             except ValueError:
                 return
-        ref = self.start if self.start.is_finite else self.stop
+        if self.start.is_infinite:
+            if self.stop.is_infinite:
+                ref = S.Zero
+            else:
+                ref = self.stop
+        else:
+            ref = self.start
         if (ref - other) % self.step:  # off sequence
             return S.false
         return _sympify(other >= self.inf and other <= self.sup)
```
gives
```python
>>> 11 in Range(-oo,oo)
True
>>> 11 in Range(10,20,2)
False
```

File: sympy/sets/fancysets.py [hunk 1]
```diff
@@ -659,8 +659,11 @@ def _contains(self, other):
                 _ = self.size  # validate
             except ValueError:
                 return
-        ref = self.start if self.start.is_finite else self.stop
-        if (ref - other) % self.step:  # off sequence
+        if self.start.is_finite:
+            ref = self.start
+        else:
+            ref = self.stop
+        if (ref - other) % self.step is S.true:  # off sequence
```
Comment (smichr) [inline sympy/sets/fancysets.py]: A mod with be a number or nan (or Mod if symbolic values are involved), not S.true.

**Comment by sylee957** [comment]
This still gives an error with the LaTeX printing.
And I also think that `Range(oo, -oo, -1)` should be canonicalized to `Range(-oo, oo)`

**Comment by ethankward** [comment]
What error with LaTeX printing do you see?

**Comment by sylee957** [comment]
`latex(Range(-oo, oo))`
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-1-649339655bf5> in <module>
      1 from sympy import *
      2 
----> 3 latex(Range(-oo, oo))

~\Documents\Github\sympy\sympy\printing\latex.py in latex(expr, fold_frac_powers, fold_func_brackets, fold_short_frac, inv_trig_style, itex, ln_notation, long_frac_ratio, mat_delim, mat_str, mode, mul_symbol, order, symbol_names, root_notation, mat_symbol_style, imaginary_unit, gothic_re_im, decimal_separator)
   2697     }
   2698 
-> 2699     return LatexPrinter(settings).doprint(expr)
   2700 
   2701 

~\Documents\Github\sympy\sympy\printing\latex.py in doprint(self, expr)
    220 
    221     def doprint(self, expr):
--> 222         tex = Printer.doprint(self, expr)
    223 
    224         if self._settings['mode'] == 'plain':

~\Documents\Github\sympy\sympy\printing\printer.py in doprint(self, expr)
    247     def doprint(self, expr):
    248         """Returns printer's representation for expr (as a string)"""
--> 249         return self._str(self._print(expr))
    250 
    251     def _print(self, expr, **kwargs):

~\Documents\Github\sympy\sympy\printing\printer.py in _print(self, expr, **kwargs)
    285                 printmethod = '_print_' + cls.__name__
    286                 if hasattr(self, printmethod):
--> 287                     return getattr(self, printmethod)(expr, **kwargs)
    288             # Unknown object, fall back to the emptyPrinter. Checks what type of
    289             # decimal separator to print.

~\Documents\Github\sympy\sympy\printing\latex.py in _print_Range(self, s)
   1905 
   1906         if s.start.is_infinite:
-> 1907             printset = dots, s[-1] - s.step, s[-1]
   1908         elif s.stop.is_infinite:
   1909             it = iter(s)

~\Documents\Github\sympy\sympy\sets\fancysets.py in __getitem__(self, i)
    827             if i == -1:
    828                 if self.stop.is_infinite:
--> 829                     raise ValueError(ooslice)
    830                 n = self.stop - self.step
    831                 if n.is_Integer or (

ValueError: cannot slice from the end with an infinite value
```

**Comment by ethankward** [comment]
The `mathml` printer doesn't give an error, but I'm not familiar with the syntax- is it correct?

```
>>> mathml(Range(-oo, oo))
'<range><apply><minus/><infinity/></apply><infinity/><cn>1</cn></range>'
```

**Comment by sylee957** [comment]
I don't think that there is a problem with mathml content,
but `mathml(Range(-oo, oo), printer='presentation')` gives an error

File: sympy/sets/fancysets.py [hunk 2]
```diff
@@ -659,9 +659,13 @@ def _contains(self, other):
                 _ = self.size  # validate
             except ValueError:
                 return
-        ref = self.start if self.start.is_finite else self.stop
-        if (ref - other) % self.step:  # off sequence
-            return S.false
+        if self.start.is_finite:
```
Comment (smichr) [inline sympy/sets/fancysets.py]: nit: we already know if `ref - other` is going to be finite by looking at self.stop:
```python
if self.start.is_finite:
    ref = self.start
elif self.stop.is_finite:
    ref = self.stop
else:
    return other.is_Integer
if (ref - other) % self.step:
    return S.false
return _sympify(other >= self.inf and other <= self.sup)
```

**Comment by smichr** [comment]
Printing a canonical infinite range without the step indicated (which can only be 1 when both ends are infinite) can be a separate issue.

**Pull request merged**